lib/types/form-types.ts

```diff
@@ -0,0 +1,148 @@
+}
+
+// the choices available for fields with multiple options like dropdowns or radio buttons
+export interface PropertyChoices {
+  en: string;
+  fr: string;
+  [key: string]: string;
+}
+
+// used to define attributes for the properties of an element in the form
+export interface ElementProperties {
+  titleEn: string;
+  titleFr: string;
+  placeholderEn?: string;
+  placeholderFr?: string;
+  descriptionEn?: string;
+  descriptionFr?: string;
+  validation?: ValidationProperties | undefined;
+  choices?: Array<PropertyChoices>;
+  subElements?: Array<FormElement>;
+  fileType?: string | undefined;
+  headingLevel?: string | undefined;
+  isSectional?: boolean;
+  maxNumberOfRows?: number;
+  [key: string]:
+    | string
+    | number
+    | boolean
+    | Array<PropertyChoices>
+    | Array<FormElement>
+    | ValidationProperties
+    | undefined;
+}
+
+// all the possible types of form elements
+export enum FormElementTypes {
+  textField = "textField",
+  textArea = "textArea",
+  dropdown = "dropdown",
+  radio = "radio",
+  checkbox = "checkbox",
+  fileInput = "fileInput",
+  richText = "richText",
+  dynamicRow = "dynamicRow",
+}
+// used to define attributes for a form element or field
+export interface FormElement {
+  id: number;
+  subId?: string;
+  type: FormElementTypes;
+  properties: ElementProperties;
+  onchange?: (event: ChangeEvent) => void;
+}
+
+/**
+ * types to define form configuration objects
+ */
+
+// defines the fields in the object that controls how form submissions are handled
+export interface SubmissionProperties {
+  email?: string;
+  vault?: boolean;
+}
+
+// defines the fields in the object that controls form branding
+export interface BrandProperties {
+  name?: string;
+  logoEn: string;
+  logoFr: string;
+  logoTitleEn: string;
+  logoTitleFr: string;
+  urlEn?: string;
+  urlFr?: string;
+}
+
+// defines the fields for the main form configuration object
+export interface FormProperties {
+  id?: string;
+  titleEn: string;
+  titleFr: string;
+  emailSubjectEn?: string;
+  emailSubjectFr?: string;
+  version: string;
+  layout: Array<string>;
+  brand?: BrandProperties;
+  elements: Array<FormElement>;
+  endPage?: Record<string, string>;
+  [key: string]:
+    | string
+    | boolean
+    | Array<string>
+    | Array<FormElement>
+    | Record<string, string>
+    | BrandProperties
+    | undefined;
+}
+
+// defines the fields for the top level form object
+export interface FormConfiguration {
+  internalTitleEn?: string;
+  internalTitleFr?: string;
+  publishingStatus: boolean;
+  submission: SubmissionProperties;
+  displayAlphaBanner?: boolean;
+  form: FormProperties;
+  securityAttribute: string;
+  reCaptchaID?: string;
+}
+
+// defines the fields for the form record that is available to unauthenticated users
+export interface PublicFormRecord {
+  formID: number;
+  formConfig: Omit<FormConfiguration, "internalTitleEn" | "internalTitleFr" | "submission">;
+  organization?: boolean;
```

Comment:
No idea as well. But it existed before on various other types so I thought against removing it. We can open another task to remove organization code since it doesn't really seem like it would ever be in use